src/drivers/github.js

```diff
@@ -288,6 +288,112 @@ class Github {
     });
   }
 
+  async pipelineJobs(opts = {}) {
+    const { ids } = opts;
+    const { owner, repo } = ownerRepo({ uri: this.repo });
+    const { actions } = octokit(this.token, this.repo);
+
+    const jobs = await Promise.all(
+      ids.map(async (id) => {
+        const { data } = await actions.getJobForWorkflowRun({
+          owner,
+          repo,
+          job_id: id
+        });
+
+        return data;
+      })
+    );
+
+    return jobs.map((job) => {
+      const { id, started_at: date, run_id: runId, status } = job;
+      return { id, date, runId, status };
+    });
+  }
+
+  async job(opts = {}) {
+    const { time, status = 'queued' } = opts;
+    const { owner, repo } = ownerRepo({ uri: this.repo });
+    const { actions } = octokit(this.token, this.repo);
+
+    const {
+      data: { workflow_runs: workflowRuns }
+    } = await actions.listWorkflowRunsForRepo({
+      owner,
+      repo,
+      status
+    });
+
+    let runJobs = await Promise.all(
+      workflowRuns.map(async (run) => {
+        const {
+          data: { jobs }
+        } = await actions.listJobsForWorkflowRun({
+          owner,
+          repo,
+          run_id: run.id,
+          status
+        });
+
+        return jobs;
+      })
+    );
+
+    runJobs = [].concat.apply([], runJobs).map((job) => {
+      const { id, started_at: date, run_id: runId } = job;
+      return { id, date, runId };
+    });
+
+    const job = runJobs.reduce((prev, curr) => {
+      const diffTime = (job) => Math.abs(new Date(job.date).getTime() - time);
+      return diffTime(curr) < diffTime(prev) ? curr : prev;
+    });
```

Comment:
Wouldn't it be a bit more obvious if we used `sort` and `pop` instead of `reduce` to get the closest match in time? Not that it matters, though.

Reply:
I assume that there isn't any other way of getting a run identifier from the self-hosted runner and using it to get the job.